src/Apps/Order/Routes/NewPayment/index.tsx

```diff
@@ -206,6 +226,12 @@ export const NewPaymentRoute: FC<NewPaymentProps & StripeProps> = props => {
   const handleFixFailedPaymentError = async (code: string) => {
     switch (code) {
       case "capture_failed": {
+        trackErrorMessageEvent(
+          "Charge failed",
+          "Payment has been declined. Please contact your card provider or bank institution, then press “Continue” again. Alternatively, use another payment method.",
+          code
+        )
```

Comment:
Ah, i was thinking of a big switch error handler like this recently. I'll bet we could make this ever more general than just the new payment step.

Reply:
Question: This message string is duplicated directly below in `dialog.showErrorDialog()`. If we want to track errors we are showing to users could a single function do both?

Reply:
Yeah, it may be time to start consolidating a lot of this tracking + dialog code. We rushed the initial batch of tracking events so that we could start seeing data about errors in the checkout flow, but this code is starting to become cumbersome.

@rquartararo and I tried to refactor the dialog to do its own data tracking early on in this project, and got blocked trying to convert the dialog into a functional component (to use the tracking hook). Again, our goal was to see data ASAP so we didn't pursue it, but now would be a good time to revisit that work.